cloud/db_cloud_impl.cc

```diff
@@ -341,28 +341,18 @@ Status DBCloudImpl::DoCheckpointToCloud(
   dbid = rtrim_if(trim(dbid), '\n');
   files_to_copy.emplace_back(IdentityFileName(""), IdentityFileName(""));
 
-  // MANIFEST file
-  auto current_epoch = cenv->GetCloudManifest()->GetCurrentEpoch();
-  auto manifest_fname = ManifestFileWithEpoch("", current_epoch);
-  auto tmp_manifest_fname = manifest_fname + ".tmp";
-  auto fs = base_env->GetFileSystem();
-  st =
-      CopyFile(fs.get(), GetName() + "/" + manifest_fname,
-               GetName() + "/" + tmp_manifest_fname, manifest_file_size, false,
-               nullptr, Temperature::kUnknown);
-  if (!st.ok()) {
-    return st;
-  }
-  files_to_copy.emplace_back(tmp_manifest_fname, std::move(manifest_fname));
-
-  // CLOUDMANIFEST file
-  files_to_copy.emplace_back(cenv->CloudManifestFile(""), cenv->CloudManifestFile(""));
-
   std::atomic<size_t> next_file_to_copy{0};
   int thread_count = std::max(1, options.thread_count);
   std::vector<Status> thread_statuses;
   thread_statuses.resize(thread_count);
 
+  auto upload_file = [&](const std::shared_ptr<CloudStorageProvider>& provider,
+                         const std::string& localName,
+                         const std::string& destName) {
+    return provider->PutCloudObject(
```

Comment:
So the issue is before a bulk leaf finishes checkpointing all the files, the shard is opened as durable/ephemeral,  which gets the CLOUDMANIFEST file but no corresponding sst files?

You mentioned that there are 32 threads uploading the sst file. I guess the `PutCloudObject` puts the file into some queue, and then these 32 threads pull the files to upload from the queue?

If that's the case, this change helps, but seems it still doesn't solve the issue completely? In the worst case, isn't it still possible that 32 threads get the final batch from the queue, which includes 30 sst files, 1 MANIFEST file and 1 CLOUDMANIFEST file, and the thread uploading the CLOUDMANIFEST file could potentially still finish before the other 31 threads?

Reply:
Yeah that's why here I only upload `MANIFEST` and `CLOUDMANIFEST` after all the other threads have been joined.

Reply:
ahh I see.